Assets/Scripts/Networking/GameManager.cs

```diff
@@ -166,6 +170,25 @@ public void QuitApplication()
 		#endregion
 
 		#region Private Methods
+		
+		void RespawnCheck()
+        {
+			GameObject[] players = GameObject.FindGameObjectsWithTag("Player");
+			//Check if any player is still alive
+			for (int i = 0; i < players.Length; i++)	
+			{
+				if (players[i].GetComponent<PlayerManager>().health > 0)
+				{
+					return;
+				}
+			}
+			//if not, respawn all players
+			for (int i = 0; i < players.Length; i++)
+			{
+				players[i].GetComponent<PlayerManager>().Respawn();
+			}
```

Comment:
Why not use `foreach` here? It's a lot better for readability and a lot better practice in real world.